test/gridfs/test.sh

```diff
@@ -28,3 +21,24 @@ else
   rdmd -g -I../../source test_gridfs.d small.bin $(sha1sum small.bin | cut -d ' ' -f 1)
   rdmd -g -I../../source test_gridfs.d large.bin $(sha1sum large.bin | cut -d ' ' -f 1)
 fi
+
+# test retrieval of many small files
+dd if=/dev/urandom of=many.bin bs=1M count=100
+split -b 100000 --additional-suffix .bin many.bin many
+for file in `ls many*.bin`; do
+    mongofiles --quiet -d test_files -r put $file >/dev/null
+done
+
+if [[ -x ./test_gridfs_many ]] ; then
+    ls many*.bin | xargs ./test_gridfs_many
+else
+    ls many*.bin | xargs rdmd -g -I../../source test_gridfs_many.d
+fi
+
+cleanup() {
+    for file in `ls *.bin`; do
+	mongofiles --quiet -d test_files delete $file >/dev/null
+	rm -f $file
+    done
+}
+trap cleanup EXIT
```

Comment:
I think moving this code to the end defeats the purpose of making it an `EXIT trap`, which calls on exit even when the script exits early because of errors

Reply:
Nice catch!